### DISCUSSION TITLE
Select 控件无法正常选择值

### DISCUSSION THREAD

**Discussion by xunmeng2002** [dotnetcore/BootstrapBlazor#3016] [Q&A] [answered]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

我使用了两个Select控件，第二个Select控件根据第一个Select的选择项，添加不同的Items集合，在测试时候发现，第一个Select选择几次后就出问题了，无法 改变选择的值。调试时候发现，改变第一个Select的 选择项时，它的OnSelectedItemChanged事件会触发两次，第一次是选择的值，第二次是null。

### Expected Behavior

Select能正确相应用户的选择

### Steps To Reproduce

[TestBootstrapBlazor.zip](https://github.com/dotnetcore/BootstrapBlazor/files/14444515/TestBootstrapBlazor.zip)

### .NET Version

net8.0

### Anything else?

运行程序后，选择左侧菜单第二个 TestSelect，然后选择右侧上面的那个Select控件，选择不同的值，来回四五次左右，该控件就无法相应用户选择了。

**Comment by bb-auto** [comment]
@xunmeng2002 Thank you for reporting. We will give triage later.

**Comment by ArgoZhang** [comment, marked as answer]
绑定的是枚举类型导致的，枚举类型内部做了特殊处理不需要 `Items`

更改代码如下：
···
<Select@bind-Value="@showType" OnSelectedItemChanged="OnSelectedItemChanged">Show/Hide</Select>
···

即可

或者集合更改为 

```
ShowTypeSourceItems.Add(new SelectedItem("ShowNone", "ShowNone"));
ShowTypeSourceItems.Add(new SelectedItem("ShowNumber", "ShowNumber"));
ShowTypeSourceItems.Add(new SelectedItem("ShowZodiac", "ShowZodiac"));
```

不支持 `int` 枚举字符串形式
